File: 181 excersises/172.py
def decimal_a_binario(decimal):
    if decimal == 0:
        return "0"
    
    bits = []
    n = decimal
    
    while n > 0:
        bits.append(str(n % 2))
        n = n // 2
    
    return ''.join(reversed(bits))

numero = int(input("Ingresa un numero decimal: "))
binario = decimal_a_binario(numero)
print(f"El numero {numero} en binario es: {binario}")

File: 181 excersises/172.cpp
#include <iostream>
using namespace std;

void decimalABinarioRecursivo(int* nPtr) {
    if (*nPtr > 1) {
        int temp = *nPtr / 2;
        decimalABinarioRecursivo(&temp);
    }
    cout << (*nPtr % 2);
}

int main() {
    int numero;
    cout << "Ingresa un numero decimal: ";
    cin >> numero;
    
    cout << "El numero " << numero << " en binario es: ";
    decimalABinarioRecursivo(&numero);
    cout << endl;
    
    return 0;
}